src/core/types/signatures/ed25519_signature.rs

```diff
@@ -1,9 +1,13 @@
 use crate::prelude::*;
 
+json_data_convertible!(Ed25519Signature);
+
 /// Represents an ED25519 signature.
 #[derive(
     Clone,
     Copy,
+    Serialize,
```

Comment:
I think this is incorrect, remove `Serialize` and `Deserialize` from `derive` and use only `json_data_convertible!(Ed25519Signature);`

Reply:
I'm getting this error if I remove them:
> json_data_convertible!(Ed25519Signature);
the trait `for<'a> interaction_id::_::_serde::Deserialize<'a>` is not implemented for `ed25519_signature::Ed25519Signature`

Reply:
@danvleju-rdx you are absolutely right, sorry! I got confused by all my macros I've written :D this one uniffi exportes it only, thus requiring it to be marked serde... but I think you should use:

```
    SerializeDisplay,
    DeserializeFromStr,
```

instead, relying on `std::fmt::Display` and `FromStr` (via our `derive_more::Display` and `derive_more::FromStr`)